test/InferOptTestUtils/loss.jl

```diff
@@ -1,56 +1,68 @@
 abstract type PipelineLoss end
 
-struct PipelineLossExperience{E,M,L} <: PipelineLoss
+struct PipelineLossExperience{E,M,L,MK,LK} <: PipelineLoss
     encoder::E
     maximizer::M
     loss::L
+    maximizer_kwargs::MK
+    loss_kwargs::LK
 end
 
-struct PipelineLossImitation{E,M,L} <: PipelineLoss
+struct PipelineLossImitation{E,M,L,MK,LK} <: PipelineLoss
     encoder::E
     maximizer::M
     loss::L
+    maximizer_kwargs::MK
+    loss_kwargs::LK
 end
 
-struct PipelineLossImitationθ{E,M,L} <: PipelineLoss
+struct PipelineLossImitationθ{E,M,L,MK,LK} <: PipelineLoss
     encoder::E
     maximizer::M
     loss::L
+    maximizer_kwargs::MK
+    loss_kwargs::LK
 end
 
-struct PipelineLossImitationθy{E,M,L} <: PipelineLoss
+struct PipelineLossImitationθy{E,M,L,MK,LK} <: PipelineLoss
     encoder::E
     maximizer::M
     loss::L
+    maximizer_kwargs::MK
+    loss_kwargs::LK
 end
 
-struct PipelineLossImitationLoss{E,M,L} <: PipelineLoss
+struct PipelineLossImitationLoss{E,M,L,MK,LK} <: PipelineLoss
     encoder::E
     maximizer::M
     loss::L
+    maximizer_kwargs::MK
+    loss_kwargs::LK
```

Comment:
This only works when passing (constant) frank wolfe kwargs because they do not change between instances.
It won't work if we have an example test using a CO algorithm with instance-dependent kwargs (it's the case in generalized maximizer tests).

Reply:
shit